contracts/eoslib/fixedpoint.hpp

```diff
@@ -0,0 +1,565 @@
+    * by passing the number to be represented in integer format and the number of decimals 
+    * required.
+    * These template classes also support the arithmetic operations and basic comparison operators
+    * 
+    */
+
+    // Some forward declarations
+    template <uint32_t q> struct fixed_point32;
+    template <uint32_t q> struct fixed_point64;
+    template <uint32_t q> struct fixed_point128;
+
+    // Will support fixed_point256 in next release    
+#if 0
+    template <uint32_t q> struct fixed_point256;
+    template <uint32_t q> 
+    struct fixed_point256
+    {
+        int128_t val;
+        fixed_point256(int256_t v=0) : val(v) {}    
+        template <uint32_t qr> fixed_point256(const fixed_point256<qr> &r);
+        template <uint32_t qr> fixed_point256(const fixed_point128<qr> &r);
+        template <uint32_t qr> fixed_point256(const fixed_point64<qr> &r);
+        template <uint32_t qr> fixed_point256(const fixed_point32<qr> &r);
+        /**
+        * Get the integer part of the 64 bit fixed number
+        * @brief To get the integer part of the fixed number
+        * @return Returns integer part of the fixed number
+        *
+        * Example:
+        * @code
+        * fixed64<18> a(1234.455667)
+        * std::cout << a.int_part(); // Output: 1234
+        * @endcode
+        */
+        int128_t int_part() const {
+            return val >> q;
+        }
+
+        /**
+        * Get the decimal part of the 64 bit fixed number
+        * @brief To get the decimal part of the fixed number
+        * @return Returns decimal part of the fixed number
+        *
+        * Example:
+        * @code
+        * fixed64<18> a(1234.455667)
+        * std::cout << a.decimal_part(); // Output: 45567
+        * @endcode
+        */
+        uint128_t frac_part() const {
+            if(!q) return 0;
+            return val << (32-q);
+        }
+
+        
+
+        template <uint32_t qr> fixed_point256 &operator=(const fixed_point32<qr> &r);
+        template <uint32_t qr> fixed_point256 &operator=(const fixed_point64<qr> &r);
+        template <uint32_t qr> fixed_point256 &operator=(const fixed_point128<qr> &r);
+        template <uint32_t qr> fixed_point256 &operator=(const fixed_point256<qr> &r);
+        // Comparison functions
+        template <uint32_t qr> bool operator==(const fixed_point256<qr> &r) { return (val == r.val);}
+        template <uint32_t qr> bool operator>(const fixed_point256<qr> &r) { return (val > r.val);}
+        template <uint32_t qr> bool operator<(const fixed_point256<qr> &r) { return (val < r.val);}
+    };
+#endif    
+
+    /**
+    * 128 bits representation of Fixed Point class
+    *
+    * Example:
+    * @code
+    * fixed_point128<6> a(123232.455667233)
+    * fixed_point128<0> a(123424)
+    * fixed_point128<18> c = a*b;
+    * fixed_point128<24> d = a+b+c;
+    * fixed_point128<24> e = b/a;
+    * @endcode
+    */
+    template <uint32_t q> 
```

Comment:
q, as a template parameter, should be capital and its meaning should be documented.

Reply:
Also it should be uint8_t and there should be a static_assert that it is less than 128